src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -28,17 +38,40 @@
     <p>Here's everything that's happened on the site so far!</p>
 
     <div id="activity">
-      <%List<User> users = 
-        (List<User>) request.getAttribute("users");
-      %>
       <ul>
-        <% 
-          for(User user : users) {
-            Date date = Date.from(user.getCreationTime());
+        <%
+        for(Map.Entry<Instant, Object> entry : sortedEventsMap.entrySet()) {
+          Date date = Date.from(entry.getKey());
         %>
-        <li><b><%= date %>: </b> 
-          <a href="/profile/<%= user.getName() %>">
-          <%= user.getName() %></a> joined!</li>
+          <li>
+            <b><%= date %>:</b>
+          <%
+          if(entry.getValue() instanceof User) {
+            User user = (User) entry.getValue();
+          %>
+            <a href="/profile/<%= user.getName() %>"><%= user.getName() %></a> 
+              joined!
+          </li>
+          <%
+          }
+          else if (entry.getValue() instanceof Conversation) {
+            Conversation conversation = (Conversation) entry.getValue();
+            User conversationOwner = null;
+            for (User user : users) {
```

Comment:
If we have the user id, can we get the user from the UserStore directly instead of searching for it linearly in the list of users?

Reply:
I wanted to do this but was unsure how to go about it. Would I create an instance of UserStore in the servlet class and set it as an attribute to the request, or create the instance in the jsp file?

Reply:
I haven't tried this so I don't know if it's doable, but can we pass the reference to the `UserStore` singleton to the jsp as an attribute?

Reply:
I'll try it out asap.